### ISSUE TITLE
FileViewFinder exception on AWS EC2

### ISSUE THREAD

**Issue by kyonkopa** [laravel/framework#19451]
After having successfully deployed Laravel using Elasticbeanstalk, Laravel throws InvalidArgumentException in FileViewFinder.php because the view path that is generated somehow still remains **/var/app/ondeck/resources/views/** which is a temporary location for the app during deployment instead of **/var/app/current/resources/views/** which is the directory where the application resides after it has been deployed.
This means the whole site is down because Laravel is trying to look for views in the wrong directory.

I couldn't find any solution to this on AWS forums. Any help?

**Comment by thecrypticace** [comment]
Perhaps the config is being cached in the first directory during deployment? If so, that can cause problems.

**Comment by kyonkopa** [comment]
Interestingly I run php artisan config:cache

**Comment by thecrypticace** [comment]
Okay, so if you run `php artisan config:cache` it needs to be done after AWS has moved the app to the new directory (unless you can setup symlinks with EC2 deployments). I'm not familiar enough with AWS to tell you how to achieve that so I'll have to defer to someone who knows.

**Comment by kyonkopa** [comment]
Great

**Comment by themsaid** [comment]
Not familiar with `Elasticbeanstalk` or deployment on AWS but @thecrypticace assumption can be true, the config is cached somewhere.

**Comment by kyonkopa** [comment]
@themsaid I think that is true because doing __DIR__ in production actually confirms that the app is being served from **/var/app/current** which is good only that the view path is wrong. I'll have to find why `cache:clear` isnt working

**Comment by kyonkopa** [comment]
After battling with it, I managed to use a post deployment script **/opt/elasticbeanstalk/hooks/appdeploy/post/artisan_clear_cache.sh** to run `php artisan config:cache`. Unfortunately this method is not documented.
1. I had to create the directory because it didnt exists by default.
2. I had to set the permissions for Laravel to write to storage.
3. Finally i had to put artisan command within the .sh.

This way the script will automatically run post deployment and clear the cache.
